pkg/virt-launcher/virtwrap/network/common.go

```diff
@@ -138,6 +154,26 @@ func (h *NetworkUtilsHandler) IptablesAppendRule(table, chain string, rulespec .
 
 	return iptablesObject.Append(table, chain, rulespec...)
 }
+func (h *NetworkUtilsHandler) NftablesNewChain(table, chain string) error {
+	output, err := exec.Command("nft", "add", "chain", "ip", table, chain).CombinedOutput()
+	if err == nil {
+		return err
+	}
+
+	log.Log.Info(string(output))
+	return nil
+
+}
+func (h *NetworkUtilsHandler) NftablesAppendRule(table, chain string, rulespec ...string) error {
+	cmd := append([]string{"add", "rule", "ip", table, chain}, rulespec...)
+	return exec.Command("nft", cmd...).Run()
+}
+func (h *NetworkUtilsHandler) NftablesNewTable(table string) error {
+	return exec.Command("nft", "add", "table", table).Run()
+}
+func (h *NetworkUtilsHandler) NftablesLoad(fnName string) error {
+	return exec.Command("nft", "-f", fmt.Sprintf("/etc/nftables/%s.nft", fnName)).Run()
```

Comment:
just curious: who creates the file inside launcher pod? why do we need to load it?

Reply:
the nftable package create the files.

this is the file we use it's a base line for the nat table equals to `iptables -t nat`
```bash
#!/usr/sbin/nft -f

table filter {
	chain input		{ type filter hook input priority 0; }
	chain forward		{ type filter hook forward priority 0; }
	chain output		{ type filter hook output priority 0; }
}
```